python/src/spark_rapids_ml/install.py

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import importlib
+import sys
+import types
+from typing import Any
+
+_accelerated_estimators = {
+    "feature": ["PCA"],
+    "clustering": ["KMeans"],
+    "classification": ["LogisticRegression", "RandomForestClassifier"],
+    "regression": ["LinearRegression", "RandomForestRegressor"],
+    "tuning": ["CrossValidator"],
+}
+
+
+_rapids_modules = {
+    module_name: importlib.import_module(f"spark_rapids_ml.{module_name}")
+    for module_name in _accelerated_estimators.keys()
+}
+_pyspark_modules = {
+    module_name: importlib.import_module(f"pyspark.ml.{module_name}")
+    for module_name in _accelerated_estimators.keys()
+}
+
+
+def set_pyspark_mod_getattr(mod_name: str) -> None:
+    proxy_module = types.ModuleType(f"pyspark.ml.${mod_name}")
+
+    def _getattr(attr: str) -> Any:
+        frame = sys._getframe()
+        assert frame.f_back
+        calling_path = frame.f_back.f_code.co_filename
+        if any(
+            [
+                (
+                    f"pyspark/ml/{mod_name}" in calling_path
+                    or f"spark_rapids_ml/{mod_name}" in calling_path
```

Comment:
Would this always be true? The calling_path must come from either pyspark or spark_rapids_ml I think.
If so, the if condition will always be true.

Reply:
No it is not always true.   the condition will only be true if the import statement or module attribute access to reach here happened in the specified modules (our modules or select pyspark/ml modules).  In these cases we want to use the pyspark.ml version of the class always.   This allows us to subclass pyspark ml classes like in our CrossValidator and also use pyspark classes to validate parameters.